File: analytics/src/main/java/org/odk/collect/analytics/NoopAnalytics.java
package org.odk.collect.analytics;

public class NoopAnalytics implements Analytics {

    @Override
    public void logEvent(String category, String action) {

    }

    @Override
    public void logEvent(String category, String action, String label) {

    }

    @Override
    public void logFormEvent(String event, String formIdHash) {

    }

    @Override
    public void logFatal(Throwable throwable) {

    }

    @Override
    public void logNonFatal(String message) {

    }

    @Override
    public void logServerEvent(String event, String serverHash) {

    }

    @Override
    public void setAnalyticsCollectionEnabled(boolean isAnalyticsEnabled) {

    }

    @Override
    public void setUserProperty(String name, String value) {

    }
}

File: analytics/src/main/java/org/odk/collect/analytics/NoopAnalytics.kt
package org.odk.collect.analytics

class NoopAnalytics : Analytics {
    override fun logEvent(event: String) {}
    override fun logEventWithParams(event: String, params: Map<String, String>) {}
    override fun logNonFatal(throwable: Throwable) {}
    override fun logMessage(message: String) {}
    override fun setAnalyticsCollectionEnabled(isAnalyticsEnabled: Boolean) {}
    override fun setUserProperty(name: String, value: String) {}
}
